presto-phoenix/src/main/java/io/prestosql/plugin/phoenix/PhoenixSplitManager.java

```diff
@@ -0,0 +1,252 @@
+import org.apache.hadoop.mapreduce.InputSplit;
+import org.apache.hadoop.mapreduce.Job;
+import org.apache.hadoop.mapreduce.JobContext;
+import org.apache.phoenix.compile.QueryPlan;
+import org.apache.phoenix.coprocessor.BaseScannerRegionObserver;
+import org.apache.phoenix.iterate.MapReduceParallelScanGrouper;
+import org.apache.phoenix.jdbc.PhoenixConnection;
+import org.apache.phoenix.jdbc.PhoenixStatement;
+import org.apache.phoenix.mapreduce.PhoenixInputSplit;
+import org.apache.phoenix.mapreduce.util.ConnectionUtil;
+import org.apache.phoenix.mapreduce.util.PhoenixConfigurationUtil;
+import org.apache.phoenix.query.HBaseFactoryProvider;
+import org.apache.phoenix.query.KeyRange;
+import org.apache.phoenix.shaded.com.google.common.base.Preconditions;
+import org.apache.phoenix.shaded.com.google.common.collect.Lists;
+import org.apache.phoenix.util.PhoenixRuntime;
+
+import javax.inject.Inject;
+
+import java.io.IOException;
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.stream.Collectors;
+
+import static io.prestosql.plugin.phoenix.PhoenixErrorCode.PHOENIX_ERROR;
+import static java.util.Objects.requireNonNull;
+
+public class PhoenixSplitManager
+        implements ConnectorSplitManager
+{
+    private static final Logger log = Logger.get(PhoenixSplitManager.class);
+
+    private final PhoenixClient phoenixClient;
+    private PhoenixMetadata phoenixMetadata;
+    private final Map<String, HostAddress> hostCache = new HashMap<>();
+
+    @Inject
+    public PhoenixSplitManager(PhoenixClient phoenixClient, PhoenixMetadataFactory phoenixMetadataFactory)
+    {
+        this.phoenixClient = requireNonNull(phoenixClient, "client is null");
+        this.phoenixMetadata = phoenixMetadataFactory.create();
+    }
+
+    @Override
+    public ConnectorSplitSource getSplits(ConnectorTransactionHandle transactionHandle, ConnectorSession session, ConnectorTableLayoutHandle layout, SplitSchedulingStrategy splitSchedulingStrategy)
+    {
+        PhoenixTableLayoutHandle layoutHandle = (PhoenixTableLayoutHandle) layout;
+        PhoenixTableHandle handle = layoutHandle.getTable();
+        try (PhoenixConnection connection = phoenixClient.getConnection()) {
+            String inputQuery = phoenixClient.buildSql(
+                    connection,
+                    handle.getCatalogName(),
+                    handle.getSchemaName(),
+                    handle.getTableName(),
+                    layoutHandle.getDesiredColumns(),
+                    layoutHandle.getTupleDomain(),
+                    phoenixMetadata.getColumns(handle, false));
+
+            return new FixedSplitSource(getHadoopInputSplits(inputQuery).stream().map(split -> (PhoenixInputSplit) split).map(split -> {
+                List<HostAddress> addresses = getSplitAddresses(split);
```

Comment:
This variable can be inlined since it's only used once. And after that, the lambda body can become a lambda expression (no need for a block with `return`)